src/main/java/com/easypost/model/StatelessRateDeserializer.java

```diff
@@ -0,0 +1,31 @@
+package com.easypost.model;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+
+import java.lang.reflect.Type;
+
+public final class StatelessRateDeserializer implements JsonDeserializer<StatelessRate[]> {
```

Comment:
Question: Why do we need a custom deserializer for Stateless Rate? Couldn't we just deserialize this to a collection, and then have the function itself return the `rates` property of that collection, rather than effectively overriding the deserialization process to start deserialization at the "rates" key?

I would imagine this is similar to the SmartRate deserialization, which also has a custom deserializer. I guess refresh my memory as to why we didn't get rid of that a few months ago? (I know I was the one who added it, but that was the first week I was here, I didn't know what I know now).

Reply:
Because the API not only returns a list of rates but also `from_address`, `to_address`, `parcel`, and `message`, so we need some tweaks for the deserialization.

Yes, creating a collection works but we will need to create another class for `option` because there isn't one. As a result, I went for a custom deserializer because we don't need to introduce two additional classes in the model and we are only deserializing what we needed.

Reply:
Fair enough, I just don't want to get to a point where we have a half-dozen custom deserializers. This is acceptable.